compass/ocean/tests/hurricane/init/interpolate_atm_forcing.py

```diff
@@ -0,0 +1,265 @@
+                ax1.set_extent([0, 359.9, -90, 90], crs=ccrs.PlateCarree())
+                ax1.add_feature(cfeature.LAND, zorder=100)
+                ax1.add_feature(cfeature.LAKES, alpha=0.5, zorder=101)
+                ax1.add_feature(cfeature.COASTLINE, zorder=101)
+                ax1.set_title('interpolated data '+time[i].strip().decode())
+                cbar = fig.colorbar(cf, ax=ax1)
+                cbar.set_label(var_label)
+
+                # Save figure
+                fig.tight_layout()
+                fig.savefig(var_abrev+'_'+str(i).zfill(4)+'.png',
+                            box_inches='tight')
+                plt.close()
+
+    def write_to_file(self, filename, data, var, xtime):
+        """
+        Write data to netCDF file
+        """
+
+        if os.path.isfile(filename):
+            data_nc = netCDF4.Dataset(filename, 'a',
+                                      format='NETCDF3_64BIT_OFFSET')
+        else:
+            data_nc = netCDF4.Dataset(filename, 'w',
+                                      format='NETCDF3_64BIT_OFFSET')
+
+            # Find dimesions
+            ncells = data.shape[1]
+            nsnaps = data.shape[0]
+
+            # Declare dimensions
+            data_nc.createDimension('nCells', ncells)
+            data_nc.createDimension('StrLen', 64)
+            data_nc.createDimension('Time', None)
+
+            # Create time variable
+            time = data_nc.createVariable('xtime', 'S1', ('Time', 'StrLen'))
+            time[:, :] = netCDF4.stringtochar(xtime)
+
+        # Set variables
+        data_var = data_nc.createVariable(var, np.float64, ('Time', 'nCells'))
+        data_var[:, :] = data[:, :]
+        data_nc.close()
+
+    def run(self):
+        """
+        Run this step of the test case
+        """
+        # Interpolation of u and v velocities
+        u_interp, u_data, xtime = self.interpolate_data_to_grid(
+            self.grid_file,
+            self.wind_file,
+            'U_GRD_L103')
+
+        # Write to NetCDF file
+        subprocess.call(['rm', self.forcing_file])
+        # check_call(['rm', self.forcing_file], logger=logger)
+
+        self.write_to_file(self.forcing_file, u_interp[2],
+                           'windSpeedU', xtime)
+        v_interp,  v_data, xtime = self.interpolate_data_to_grid(
+            self.grid_file,
+            self.wind_file,
+            'V_GRD_L103')
+        self.write_to_file(self.forcing_file, v_interp[2],
+                           'windSpeedV', xtime)
+
+        vel_interp = u_interp
+        vel_interp[2][:] = np.sqrt(np.square(u_interp[2][:]) +
+                                   np.square(v_interp[2][:]))
+        vel_data = u_data
+        vel_data[2][:] = np.sqrt(np.square(u_data[2][:]) +
+                                 np.square(v_data[2][:]))
+        self.plot_interp_data(vel_data, vel_interp,
+                              'velocity magnitude', 'vel', xtime)
```

Comment:
It might be necessary to iterate over one of the 3 dimensions here (probably time) and compute the velocity magnitude separately for each time index.  Slower but much less memory intensive.

If you were to do this with `xarray` instead, you could use its chunking in time to handle this.